hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/NonpartitionedKeyGenerator.java

```diff
@@ -51,6 +61,24 @@ public String getRecordKey(GenericRecord record) {
     return nonpartitionedAvroKeyGenerator.getRecordKey(record);
   }
 
+  @Override
+  public String getRecordKey(Row row) {
```

Comment:
shouldn't we migrate this fix to SimpleKeyGen, if you feel existing impl in SimpleKeyGen could be fixed? why making changes just to NonPartitionedKeyGen only.

Reply:
This is addressed in #5523